instrumentation/opentelemetry-instrumentation-redis/tests/test_redis.py

```diff
@@ -311,3 +314,32 @@ def test_attributes_unix_socket(self):
             span.attributes[SpanAttributes.NET_TRANSPORT],
             NetTransportValues.OTHER.value,
         )
+
+
+class TestRedisAsync(TestBase, IsolatedAsyncioTestCase):
+    def setUp(self):
+        super().setUp()
+        RedisInstrumentor().instrument(tracer_provider=self.tracer_provider)
+
+    def tearDown(self):
+        super().tearDown()
+        RedisInstrumentor().uninstrument()
+
+    @pytest.mark.asyncio
+    async def test_redis_operations(self):
+        async def redis_operations():
+            try:
+                redis_client = FakeRedis()
+                async with redis_client.pipeline(transaction=False) as pipe:
+                    await pipe.watch("a")
+                    await redis_client.set("a", "bad")
+                    pipe.multi()
+                    await pipe.set("a", "1")
+                    await pipe.execute()
+            except WatchError:
+                pass
+
+        await redis_operations()
+
+        spans = self.memory_exporter.get_finished_spans()
```

Comment:
Are we expecting only one span here? If so, it's okay to assert the length of spans.

Reply:
you are right, we should have 3 spans, I will add more assert here